### DISCUSSION TITLE
Getting historic workflows of Github Actions

### DISCUSSION THREAD

**Discussion by thedarkwriter** [community/community#25037] [Actions]
<p>Hopefully I’m not missing something obvious, but is there a way to scroll back or call Github Action’s API to get the workflow and logs of older runs? It seems the webpage lets you look back around 12 previous workflows. If this is not a capability yet, it would be a really helpful addition.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hey <a class="mention" href="/thedarkwriter">@thedarkwriter</a>,</p>
<p>Thanks for being here!  In the UI via the Actions Tab, it’s possible to look at previously executed workflows. I can see how this can be valuable and will add it to the feedback for the team.</p>
<p>In the API, there’s no endpoint for fetching workflows. However, because GitHub Actions creates a check suite per workflow and a check run per executed action in said workflow, someone could use the Checks API to get that information though it’s not as straightforward because it requires the knowledge of a specific SHA:</p>
<p><a href="https://developer.github.com/v3/checks/" rel="nofollow noopener">https://developer.github.com/v3/checks/</a></p>
<p>I hope this helps!</p>

**Comment by thedarkwriter** [comment]
<p>Hi Andrea. Thanks for the reply. </p>
<p>I see the previous executed runs, there just aren’t very many available and no way to scroll pages back to look at previous pages of runs. For folks wanting to use this for CICD (like I’d like to) it would be a very helpful feature.</p>

**Comment by chrispat** [comment]
<p>The Actions API beta shipped today <a href="https://developer.github.com/v3/actions/" rel="nofollow noopener">https://developer.github.com/v3/actions/</a></p>
